vm-virtio/src/vhost_user/vu_common_ctrl.rs

```diff
@@ -100,8 +100,8 @@ pub fn setup_vhost_user(
     queue_evts: Vec<EventFd>,
     acked_features: u64,
 ) -> Result<Vec<(EventFd, Queue)>> {
-    // Set features based on the acked features from the guest driver.
-    vu.set_features(acked_features)
+    let backend_features = vu.get_features().unwrap();
+    vu.set_features(acked_features & backend_features)
```

Comment:
I think @sboeuf's point is that acked_features should already be the set of features supported by both the guest kernel and the backend. Perhaps you need to address this somewhere else? In the vhost-user-net code itself?

Reply:
Exactly, there is no point in masking acked_features one more time with the backend_features. The sequence is as follows:
- the virtio device emulation knows the set of available_features it can support.
- the vmm gets the backend_features.
- based on the backend_features, the vmm adjusts the set of available_features.
- then when the guest driver kicks in, it reads the available_features and set the acked_features based on what it's capable of.
- once all this is done, the activate() function is invoked and the set of features that are actually supported needs to be updated based on the acked_features.

@bjzhjing where is the problem in the ovs+dpdk case?

Reply:
@sboeuf @rbradford
Yes, the sequence is right in overall. Regarding specific implementation, like vhost-user-net, there are some subtle but critical difference. Features like VIRTIO_NET_F_MAC will tell guest OS to get mac address from config space filled by VMM device, so it should be covered in acked_features, the minimal feature set between driver and device. However, since it's not included in the backend_features from OVS+DPDK, it should not be enabled in available_features before the first adjustment with backend_features to avoid being turned off. Eventually, the backend_features should be adjusted with acked_features to remove features like VIRTIO_NET_F_MAC, and then set back by set_features call. 

It seems more sensible to fix this issue in vhost-user-net implementation, rather than in vu_common_ctrl, if it's not the case for other vhost-user devices. I will make the change.